torch/__init__.py

```diff
@@ -404,9 +406,19 @@ def set_deterministic(d):
     """
     _C._set_deterministic(d)
 
+def set_deterministic(d):
+    r"""This function is deprecated and will be removed in a future release.
+    Please use :func:`torch.use_deterministic_algorithms` instead.
+    """
+    warnings.warn((
+        "torch.set_deterministic is deprecated and will be removed in a future "
+        "release. Please use torch.use_deterministic_algorithms instead"))
+
+    use_deterministic_algorithms(d)
+
 def is_deterministic():
```

Comment:
Should we revise the name of this function, too? Like `is_deterministic_algorithms_enabled()`?

Reply:
Yeah that's probably a good idea. Maybe `is_using_deterministic_algorithms` would be a bit more consistent

Reply:
I think "enabled" is slightly more consistent with existing functions:

https://pytorch.org/docs/master/amp.html#torch.cuda.amp.GradScaler.is_enabled
https://pytorch.org/docs/master/backends.html#torch.backends.cudnn.enabled
https://pytorch.org/docs/master/autograd.html?highlight=enabled#torch.autograd.set_grad_enabled

Reply:
Ah ok, sounds good to me